### ISSUE TITLE
Collect infrastructure services health/stats to master

### ISSUE THREAD

**Issue by jakolehm** [kontena/kontena#2062]
We need to collect infrastructure service stats to master so that we can report them via api to users.

Services that need reporting:
- etcd 
- weave
- ipam
- agent

**Comment by xtagon** [comment]
A realtime websocket or pub/sub API would be really cool too.

**Comment by jnummelin** [comment]
Collecting the "requirements"

## CLI

New command `kontena grid health` that reports the statuses of all "infrastructure" services

## Master

New API for grid health, `/v1/grids/:name/health`

Model where to store the health statuses. As each of the components have different details on the health status, might be bit tricky to have one single model. One option would be to use model per component, maybe even embedded into grid model?

## Agent

Periodic (maybe 60s by default?) worker to report health statuses.

### Etcd

We need etcd status per node and also `etcdctl cluster-health` type of report

### Weave

`weave status`
`weave status connections`?

Something else?

### Ipam

Currently there's no status endpoint in Ipam, see https://github.com/kontena/kontena-ipam/issues/54

### cAdvisor

How to determine cAdvisor health? Maybe something like `cadvisor_container.running? && (get 'http://127.0.0.1:8989/api/v1.2/docker/').status == 200`

### Docker

How to determine Docker health? Would `Docker::Container.get(all: true).size > 0` be enough?

### Agent

What bits we want to collect?

**Comment by jnummelin** [comment]
I don't think we have clear enough goals set for this to make it into 1.3 release train.

**Comment by jakolehm** [comment]
Agent should at least have time sync as health "source".
